front/src/modules/timetableItem/components/Timetable/RoundTrips/utils.ts

```diff
@@ -38,41 +46,158 @@ const getStepLabels = (
-      stops: stepLabels.slice(1, -1),
-      destination: stepLabels.at(-1)!,
-      startTime: new Date(item.start_time),
-      requestedArrivalTime,
-      status: 'todo',
-    };
+export const formatPairingItems = (
+  roundTripGroups: TimetableItemRoundTripGroups,
+  t: TFunction<'operational-studies', 'main'>
+): PairingItem[] => {
+  const todoItems = roundTripGroups.others.map((item) => formatBasePairingItem(item, 'todo', t));
+  const oneWayItems = roundTripGroups.oneWays.map((item) =>
+    formatBasePairingItem(item, 'oneWays', t)
+  );
+  const roundTripItems = roundTripGroups.roundTrips.map(([itemA, itemB]) => {
+    const formattedItemA = formatBasePairingItem(itemA, 'roundTrips', t);
+    const formattedItemB = formatBasePairingItem(itemB, 'roundTrips', t);
+    const isValidPair = checkRoundTripCompatible(itemA, itemB);
+    return [
+      {
+        ...formattedItemA,
+        pairedItemId: formattedItemB.id,
+        isValidPair,
+      },
+      {
+        ...formattedItemB,
+        pairedItemId: formattedItemA.id,
+        isValidPair,
+      },
+    ];
   });
+
+  return [...todoItems, ...oneWayItems, ...roundTripItems]
+    .flat()
+    .sort((a, b) => a.name.toLowerCase().localeCompare(b.name.toLowerCase()));
 };
 
-export default formatPairingItems;
+const getItemInitialStatus = ({
+  itemRawId,
+  roundTrips,
+}: {
+  itemRawId: number;
+  roundTrips: RoundTrips;
+}) => {
+  let initialStatus: 'oneWays' | 'roundTrips' | 'todo' = 'todo';
+  if (roundTrips.one_ways?.includes(itemRawId)) {
+    initialStatus = 'oneWays';
+  } else if (roundTrips.round_trips?.flat().includes(itemRawId)) {
+    initialStatus = 'roundTrips';
+  }
+  return initialStatus;
+};
+
+export const buildRoundTripsPayload = (
+  pairingItems: PairingItem[],
+  trainScheduleRoundtrips: RoundTrips,
+  pacedTrainRoundtrips: RoundTrips
+) => {
+  const trainScheduleIdsToDelete: number[] = [];
+  const trainScheduleOneWaysIds: number[] = [];
+  const trainScheduleRoundTripsIds: number[][] = [];
+  const pacedTrainIdsToDelete: number[] = [];
+  const pacedTrainOneWaysIds: number[] = [];
+  const pacedTrainRoundTripsIds: number[][] = [];
+
+  for (const item of pairingItems) {
+    if (isTrainScheduleId(item.id)) {
+      const itemRawId = extractEditoastIdFromTrainScheduleId(item.id);
+      const initialStatus = getItemInitialStatus({
+        itemRawId,
+        roundTrips: trainScheduleRoundtrips,
+      });
+
+      if (
+        item.status === 'roundTrips' &&
+        initialStatus !== item.status &&
+        !trainScheduleRoundTripsIds.flat().includes(itemRawId)
+      ) {
+        const pairedItemRawId = isTrainScheduleId(item.pairedItemId)
+          ? extractEditoastIdFromTrainScheduleId(item.pairedItemId)
+          : extractEditoastIdFromPacedTrainId(item.pairedItemId);
+        trainScheduleRoundTripsIds.push([itemRawId, pairedItemRawId]);
+      }
+      if (item.status === 'oneWays' && initialStatus !== item.status) {
```

Comment:
If you have used the condition initialStatus !== item.status, you can simplify these tests and use else (same for below)
```suggestion
      else if (item.status === 'oneWays') {
```